pymove/tests/preprocessing/test_filters.py

```diff
@@ -0,0 +1,567 @@
+                               1.3208180727070074, nan]
+    move_df['dist_prev_to_next'] = [nan, 140440.86267282354, 7.563335999941849,
+                                    140461.50100279532, nan]
+    cols = ['id', 'lat', 'lon', 'datetime', 'dist_to_prev', 'dist_to_next',
+            'dist_prev_to_next']
+    return move_df[cols], cols
+
+
+def _prepare_df_with_dist_time_speed():
+    move_df = MoveDataFrame(
+        data=list_data_2, latitude='lat', longitude='lon', datetime='datetime'
+    )
+    move_df['dist_to_prev'] = [nan, 13.884481484192932, 140454.64510608764,
+                               140460.97747220137, 1.3208180727070074]
+    move_df['time_to_prev'] = [nan, 1.0, 5.0, 5.0, 5.0]
+    move_df['speed_to_prev'] = [nan, 13.884481484192932, 28090.92902121753,
+                                28092.195494440275, 0.26416361454140147]
+    cols = ['id', 'lat', 'lon', 'datetime', 'dist_to_prev', 'time_to_prev',
+            'speed_to_prev']
+    return move_df[cols], cols
+
+
+def _prepare_df_with_long_trajectory():
+    move_df = MoveDataFrame(
+        data=list_data_3, latitude='lat', longitude='lon', datetime='datetime'
+    )
+    move_df['dist_to_prev'] = [
+        nan, 13.884481484192932, 7.563335999941849,
+        1.3208180727070074, nan, 2039.4197149375298
+    ]
+    move_df['time_to_prev'] = [
+        nan, 61.00000000000001, 70.0,
+        65.0, nan, 5.0
+    ]
+    move_df['speed_to_prev'] = [
+        nan, 0.22761445056053983, 0.10804765714202641,
+        0.020320278041646267, nan, 407.883942987506
+    ]
+    cols = ['id', 'lat', 'lon', 'datetime', 'dist_to_prev', 'time_to_prev',
+            'speed_to_prev']
+    return move_df[cols], cols
+
+
+def test_by_bbox():
+    move_df, cols = _prepare_df_default(list_data_1)
+    bbox = (39.984193, 116.31924, 39.984222, 116.319405)
+
+    filter_values = filters.by_bbox(move_df, bbox)
+    expected = DataFrame(data=[
+        [39.984199952392578, 116.31932067871094, Timestamp('2008-10-23 05:53:06'), 1]
+    ], columns=cols, index=[1]
+    )
+    assert_frame_equal(filter_values, expected)
+    assert move_df.len() == 4
+
+    filter_out_values = filters.by_bbox(move_df, bbox, filter_out=True)
+    expected = DataFrame(data=[
+        [39.984092712402344, 116.31923675537101, Timestamp('2008-10-23 05:53:05'), 1],
+        [39.984222412109375, 116.31940460205078, Timestamp('2008-10-23 05:53:11'), 2],
+        [39.984222412109375, 116.31940460205078, Timestamp('2008-10-23 05:53:11'), 2]
+    ], columns=cols, index=[0, 2, 3]
+    )
+    assert_frame_equal(filter_out_values, expected)
+    assert move_df.len() == 4
+
+    filters.by_bbox(move_df, bbox, inplace=True)
+    expected = DataFrame(data=[
+        [39.984199952392578, 116.31932067871094, Timestamp('2008-10-23 05:53:06'), 1]
+    ], columns=cols, index=[1]
+    )
+    assert_frame_equal(move_df, expected)
+    assert move_df.len() == 1
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5574862)